mobile/src/context/BookContext.tsx

```diff
@@ -1,191 +1,57 @@
-      {
-        id: 'f',
-        title: 'bar',
-        image: 'https://placekitten.com/200/305',
-        createdAt: '1000',
-      },
-      {
-        id: 'g',
-        title: 'bar',
-        image: 'https://placekitten.com/200/306',
-        createdAt: '1000',
-      },
-      {
-        id: 'h',
-        title: 'bar',
-        image: 'https://placekitten.com/200/307',
-        createdAt: '1000',
-      },
-      {
-        id: 'i',
-        title: 'bar',
-        image: 'https://placekitten.com/200/310',
-        createdAt: '1000',
-      },
-      {
-        id: 'j',
-        title: 'foo',
-        image: 'https://placekitten.com/200/300',
-        createdAt: '10000',
-      },
-      {
-        id: 'k',
-        title: 'bar',
-        image: 'https://placekitten.com/200/301',
-        createdAt: '1500',
-      },
-      {
-        id: 'l',
-        title: 'bar',
-        image: 'https://placekitten.com/200/302',
-        createdAt: '2000',
-      },
-      {
-        id: 'm',
-        title: 'bar',
-        image: 'https://placekitten.com/200/303',
-        createdAt: '3000',
-      },
-      {
-        id: 'n',
-        title: 'bar',
-        image: 'https://placekitten.com/200/304',
-        createdAt: '1000',
-      },
-      {
-        id: 'o',
-        title: 'bar',
-        image: 'https://placekitten.com/200/305',
-        createdAt: '1000',
-      },
-      {
-        id: 'p',
-        title: 'bar',
-        image: 'https://placekitten.com/200/306',
-        createdAt: '1000',
-      },
-      {
-        id: 'q',
-        title: 'bar',
-        image: 'https://placekitten.com/200/307',
-        createdAt: '1000',
-      },
-      {
-        id: 'r',
-        title: 'bar',
-        image: 'https://placekitten.com/200/310',
-        createdAt: '1000',
-      },
-      {
-        id: 's',
-        title: 'foo',
-        image: 'https://placekitten.com/200/300',
-        createdAt: '10000',
-      },
-      {
-        id: 't',
-        title: 'bar',
-        image: 'https://placekitten.com/200/301',
-        createdAt: '1500',
-      },
-      {
-        id: 'u',
-        title: 'bar',
-        image: 'https://placekitten.com/200/302',
-        createdAt: '2000',
-      },
-      {
-        id: 'v',
-        title: 'bar',
-        image: 'https://placekitten.com/200/303',
-        createdAt: '3000',
-      },
-      {
-        id: 'w',
-        title: 'bar',
-        image: 'https://placekitten.com/200/304',
-        createdAt: '1000',
-      },
-      {
-        id: 'x',
-        title: 'bar',
-        image: 'https://placekitten.com/200/305',
-        createdAt: '1000',
-      },
-
-    ];
-    dispatch({ type: 'BOOKS_LOADED', payload: books });
+    client.getBooks().then((res) => {
+      dispatch({ type: 'BOOKS_LOADED', payload: res, loading: false });
+    }).catch((err) => {
+      console.log(err);
+      // if books not found, continue indicating loading
+      dispatch({ type: 'ERROR', payload: null, loading: true });
```

Comment:
I don't think we need to dispatch at all here, we can just keep the current state of `loading: true`. This just adds another state to the books state machine (and believe me, it's going to get messy once we add offline mode).